contrib/objectstore/subscriber/objectstore.go

```diff
@@ -54,45 +76,104 @@ func (s *Subscriber) OnStructMessage(c ws.Speaker, msg *ws.StructMessage) {
 	}
 }
 
-// StoreFlows writes a set of flows to the object storage service
+// StoreFlows store flows in memory, before being written to the object store
 func (s *Subscriber) StoreFlows(flows []*flow.Flow) error {
-	if len(flows) == 0 {
-		return nil
+	endTime := time.Now()
+
+	s.flowsMutex.Lock()
+	defer s.flowsMutex.Unlock()
+
+	// transform flows and save to in-memory arrays, based on tag
+	if flows != nil {
```

Comment:
is there any execution path which may yield `flows == nil` in this line? if not then you can eliminate this if.

Reply:
We call it with a `nil` argument by a `time.AfterFunc` invocation.

Reply:
ok